cmd/kafka-example/main.go

```diff
@@ -55,80 +67,112 @@ func main() {
 	signal.Notify(signals, os.Interrupt)
 	stdinChannel := make(chan string)
 
+	eventLoopContext, eventLoopCancel := context.WithCancel(context.Background())
+	eventLoopDone := make(chan bool)
+	consumeCount := 0
+
 	go func(ch chan string) {
 		reader := bufio.NewReader(os.Stdin)
 		for {
 			line, err := reader.ReadString('\n')
 			if err != nil {
 				break
 			}
+			if cfg.Chomp {
+				line = line[:len(line)-1]
+			}
 			ch <- line
 		}
+		eventLoopCancel()
+		<-eventLoopDone
 		close(ch)
 	}(stdinChannel)
 
-	eventLoopContext, eventLoopCancel := context.WithCancel(context.Background())
-	eventLoopDone := make(chan bool)
+	// eventLoop
 	go func() {
 		defer close(eventLoopDone)
 		for {
 			select {
+			case <-time.After(1 * time.Second):
+				log.Trace("[KAFKA-TEST] tick", nil)
 			case <-eventLoopContext.Done():
+				log.Trace("[KAFKA-TEST] Event loop context done", log.Data{"eventLoopContextErr": eventLoopContext.Err()})
 				return
 			case consumedMessage := <-consumer.Incoming():
-				log.Info("[KAFKA-TEST] Received message", nil)
+				consumeCount++
+				log.Info("[KAFKA-TEST] Received message", log.Data{"consumeCount": consumeCount, "consumeMax": cfg.ConsumeMax})
 				consumedData := consumedMessage.GetData()
+				sleep := 0 * time.Millisecond
+				// sleep = 500*time.Millisecond + cfg.TimeOut
```

Comment:
I don't know whether its worth leaving a comment to explain why you would switch out sleep of 0 seconds to the timeOut value plus half a second?